apps/storybook/src/Dialog.stories.tsx

```diff
@@ -0,0 +1,270 @@
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const closeDialog = () => {
+    setIsOpen(false);
+  };
+
+  const onClose = (event: React.SyntheticEvent<Element, Event>) => {
+    action('onClose', { depth: 1 })(event);
+    closeDialog();
+  };
+
+  const primaryButtonHandle = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    action('Primary button')(event);
+    closeDialog();
+  };
+
+  const secondaryButtonHandle = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    action('Secondary button')(event);
+    closeDialog();
+  };
+
+  return (
+    <>
+      <Button styleType='high-visibility' onClick={() => setIsOpen(true)}>
+        Open Dialog
+      </Button>
+      <Dialog
+        isOpen={isOpen}
+        onClose={onClose}
+        closeOnEsc
+        isDismissible
+        isDraggable
+      >
+        <Dialog.Main>
+          <Dialog.TitleBar titleText='Best dialog ever' />
+          <Dialog.Content>
+            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+            eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim
+            ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+            aliquip ex ea commodo consequat. Duis aute irure dolor in
+            reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
+            pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
+            culpa qui officia deserunt mollit anim id est laborum.
+          </Dialog.Content>
+          <Dialog.ButtonBar>
+            <Button styleType='high-visibility' onClick={primaryButtonHandle}>
+              Primary
+            </Button>
+            <Button onClick={secondaryButtonHandle}>Secondary</Button>
+          </Dialog.ButtonBar>
+        </Dialog.Main>
+      </Dialog>
+    </>
+  );
+};
+
+export const DraggableRelativeToContainer: Story = ({ isOpen, ...rest }) => {
+  const [isDialogOpen, setIsDialogOpen] = React.useState(isOpen);
+
+  React.useEffect(() => {
+    setIsDialogOpen(isOpen);
+  }, [isOpen]);
+
+  const closeDialog = () => {
+    setIsDialogOpen(false);
+  };
+
+  const onClose = (event: React.SyntheticEvent<Element, Event>) => {
+    action('onClose', { depth: 1 })(event);
+    closeDialog();
+  };
+
+  const primaryButtonHandle = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    action('Primary button')(event);
+    closeDialog();
+  };
+
+  const secondaryButtonHandle = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    action('Secondary button')(event);
+    closeDialog();
+  };
+
+  return (
+    <>
+      <Button styleType='high-visibility' onClick={() => setIsDialogOpen(true)}>
+        Open Dialog
+      </Button>
+      <DialogWrapper
+        style={{
+          width: '70vw',
+          height: '70vh',
+          border: '1px solid red',
+          marginTop: 12,
+        }}
```

Comment:
Looking at the dialog wrapper css, it looks like it includes `top`, `left`, `width` and `height` already. Do you remember the reasoning for that? Maybe we intended for the wrapper to be `position: absolute` that the user would place inside their own container?

Reply:
Nope, I don't remember. Aren't you the author of these dialog css changes? 😁

Reply:
I am not the author, these changes came from before (all i did is rename -backdrop to -wrapper). But looking at the html, it seems like we expect the user to provide their own container (with custom sizing) and then put this wrapper inside it (with no sizing).
https://github.com/iTwin/iTwinUI/blob/main/packages/itwinui-css/backstop/tests/dialog.html#L261